lib/tumblr.js

```diff
@@ -454,7 +502,22 @@ TumblrClient.prototype.addPostMethods = function(methods) {
  */
 module.exports = {
     Client: TumblrClient,
-    createClient: function(credentials, baseUrl) {
-        return new TumblrClient(credentials, baseUrl);
+    createClient: function(credentials, returnPromises, baseUrl, requestLibrary) {
```

Comment:
How do you feel about having `options` hash here instead of a number of arguments? I feel like that if you will need to add more things here, it will become harder to manage. It's also good to do it now, while you have ability to introduce it with less-breaking change, since currently it's just two arguments.